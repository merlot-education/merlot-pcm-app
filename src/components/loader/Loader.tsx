import React from 'react'
import { Modal, View, ActivityIndicator, StyleSheet } from 'react-native'
import { ColorPallet } from '../../theme/theme'

interface LoaderProps {
  loading?: boolean
}

const styles = StyleSheet.create({
  activityIndicatorWrapper: {
    backgroundColor: 'rgba(52, 52, 52, 0.8)',
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
})

const Loader: React.FC<LoaderProps> = ({ loading }) => {
  return (
    <Modal testID="loader" animationType="none" visible={loading} transparent>
      <View style={styles.activityIndicatorWrapper}>
        <ActivityIndicator size="small" color={ColorPallet.brand.primary} />
      </View>
    </Modal>
  )
}

export default Loader
