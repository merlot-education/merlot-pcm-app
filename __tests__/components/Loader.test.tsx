import React from 'react'
import { create } from 'react-test-renderer'

import Loader from '../../src/components/loader/Loader'

describe('Loader Component', () => {
  it('Loader renders correctly', () => {
    const tree = create(<Loader />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
