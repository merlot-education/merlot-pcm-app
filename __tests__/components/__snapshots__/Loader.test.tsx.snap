// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component Loader renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(52, 52, 52, 0.8)",
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <ActivityIndicator
      color="#1E9BF2"
      size="small"
    />
  </View>
</Modal>
`;
